okonomiyaki/file_formats/_egg_info.py

```diff
@@ -142,6 +144,12 @@ def parse_rawspec(spec_string):
             res[key] = spec[key]
         except KeyError:
             raise InvalidMetadataField(key, InvalidMetadataField.undefined)
+
+    for k, v in res.items():
+        if isinstance(v, six.binary_type):
+            res[k] = decode_if_needed(v)
```

Comment:
for the comprehension, still works w/ 2.6 so no dict comprehension :( In general, I don't find `dict((k, v) for k, v....)` clearer than a for loop in this case.